### ISSUE TITLE
Fix incorrect Gaussian kernel generation (Asymmetry & Hard Edges)

### ISSUE THREAD

**Issue by nimzodisaster** [rohitrango/FireANTs#61]
I found two logical errors in make_gaussian_kernel in cc.py that negatively impact registration accuracy, particularly for LocalNormalizedCrossCorrelationLoss:

    Asymmetry ("Off-Center"): The gaussian_1d function calculates the kernel tail length as tail = sigma * truncated. It passes pass truncated = kernel_size // 2. This results in a generated array that is much larger than intended. The subsequent slice kernel[:kernel_size] discards the right half of the distribution (including the peak in some cases), resulting in a kernel that shifts the image content during convolution.

    Incorrect Sigma Scaling ("Hard Edge"): sigma is currently set to kernel_size / 3.0. This sets the full width to 3σ. Standard Gaussian approach sets the half-width r to ≈3σ to ensure the values decay to ~ zero at the edges. The current implementation creates a "mound" with hard edges, introducing ringing artifacts and high-frequency noise in gradients.

I am currently trying this patch out

```
def make_gaussian_kernel(kernel_size: int) -> torch.Tensor:
-    sigma = torch.tensor(kernel_size / 3.0)
-    kernel = gaussian_1d(sigma=sigma, truncated=(kernel_size // 2) * 1.0, approx="sampled", normalize=False) * (
-        2.5066282 * sigma
-    )
-    return kernel[:kernel_size]
+    if kernel_size % 2 == 0:
+        raise ValueError("kernel_size must be odd for Gaussian kernel")
+
+    # Sigma should be defined such that radius approx 3 standard deviations
+    radius = kernel_size // 2
+    sigma = torch.tensor(radius / 3.0)
+    
+    # Truncated is the number of sigmas, not pixels. 
+    # tail = sigma * truncated = (radius/3) * 3 = radius
+    kernel = gaussian_1d(sigma=sigma, truncated=3.0, approx="sampled", normalize=False) * (
+        2.5066282 * sigma
+    )
+    
+    # Handle potential rounding errors in gaussian_1d tail calculation
+    if kernel.shape[0] > kernel_size:
+        start = (kernel.shape[0] - kernel_size) // 2
+        kernel = kernel[start : start + kernel_size]
+        
+    return kernel
```
